CondTools/SiStrip/plugins/SiStripBadChannelPatcher.cc

```diff
@@ -0,0 +1,228 @@
+  for (const auto& id : detids) {
+    // check on the detids to exclude
+    if (std::count(detIdsToExclude_.begin(), detIdsToExclude_.end(), id)) {
+      edm::LogInfo("SiStripBadChannelPatcher") << "I AM GOING TO EXCLUDE DETID: " << id;
+      continue;
+    } else {
+      LogDebug("SiStripBadChannelPatcher") << "I AM GOING TO KEEP DETID: " << id;
+    }
+
+    // check on the FEDs to exclude
+    const auto& currentFED = this->fedFromDetId(id);
+    if (std::count(fedsToExclude_.begin(), fedsToExclude_.end(), currentFED)) {
+      edm::LogInfo("SiStripBadChannelPatcher") << "I AM GOING TO EXCLUDE DETID: " << id;
+      continue;
+    } else {
+      LogDebug("SiStripBadChannelPatcher") << "I AM GOING TO KEEP DETID: " << id;
+    }
+
+    SiStripBadStrip::Range range = payload.getRange(id);
+    std::vector<unsigned int> theSiStripVector;
+    unsigned int theBadStripRange;
+    for (std::vector<unsigned int>::const_iterator badStrip = range.first; badStrip != range.second; ++badStrip) {
+      unsigned short firstBadStrip = payload.decode(*badStrip).firstStrip;
+      unsigned short NconsecutiveBadStrips = payload.decode(*badStrip).range;
+      theBadStripRange = obj->encode(firstBadStrip, NconsecutiveBadStrips);
+      theSiStripVector.push_back(theBadStripRange);
+    }
+    SiStripBadStrip::Range outRange(theSiStripVector.begin(), theSiStripVector.end());
+    if (!obj->put(id, outRange))
+      edm::LogError("SiStripBadChannelPatcher") << "[SiStripBadChannelPatcher::analyze] detid already exists";
+  }  // loop on the detids of the original payload
+
+  const auto& detInfo =
+      SiStripDetInfoFileReader::read(edm::FileInPath{SiStripDetInfoFileReader::kDefaultFile}.fullPath());
+
+  // add to the list of DetIds to include also the one from the list of FEDs
+  const std::vector<uint32_t>& allDetIds = detInfo.getAllDetIds();
+  this->addDetIdsFromExcludedFEDs(allDetIds);
+
+  // add more full bad detids
+  if (!detIdsToInclude_.empty()) {
+    edm::LogInfo("SiStripBadChannelPatcher") << "I AM GOING TO ADD MORE DETIDS";
+
+    std::stringstream ss;
+    for (const auto& detid : detIdsToInclude_) {
+      edm::LogInfo("SiStripBadChannelPatcher") << "I AM GOING TO ADD DETID: " << detid;
+      const auto& nAPVs = detInfo.getNumberOfApvsAndStripLength(detid).first;
+
+      std::vector<unsigned int> theSiStripVector;
+      unsigned short firstBadStrip{0}, NconsecutiveBadStrips{0};
+      unsigned int theBadStripRange;
+
+      for (unsigned int n = 0; n < nAPVs; n++) {
+        firstBadStrip = n * sistrip::STRIPS_PER_APV;
+        NconsecutiveBadStrips = sistrip::STRIPS_PER_APV;
+        theBadStripRange = obj->encode(firstBadStrip, NconsecutiveBadStrips);
+
+        ss << "detid " << detid << " \t"
+           << " firstBadStrip " << firstBadStrip << "\t "
+           << " NconsecutiveBadStrips " << NconsecutiveBadStrips << "\t "
+           << " packed integer " << std::hex << theBadStripRange << std::dec << std::endl;
```

Comment:
```suggestion
        if (printDebug_) {
          firstBadStrip = n * sistrip::STRIPS_PER_APV;
          nConsecutiveBadStrips = sistrip::STRIPS_PER_APV;
          theBadStripRange = obj->encode(firstBadStrip, NconsecutiveBadStrips);
          ss << "detid " << detid << " \t"
              << " firstBadStrip " << firstBadStrip << "\t "
              << " nConsecutiveBadStrips " << nConsecutiveBadStrips << "\t "
              << " packed integer " << std::hex << theBadStripRange << std::dec << std::endl;
          }
```

Reply:
I don't understand this suggestion.
`theBadStripRange` is needed to encode which channel to mask in every case, not only when `printDebug_` is `true`.

Reply:
> I don't understand this suggestion. `theBadStripRange` is needed to encode which channel to mask in every case, not only when `printDebug_` is `true`.

Indeed, you are right.
That theBadStripRange mustang be kept out of the if block